### PR TITLE
rename plugin manifest

### LINKED ISSUE #27517: Plugin systems overloaded use of the term "manifest" is pathological
Use of the term "manifest" in the plugin system confusing the context of distribution. A manifest is an abstract of list of resources, defined in https://github.com/docker/distribution/blob/master/docs/spec/manifest-v2-2.md. Overloading the terminology for plugins will cause confusion between the encompassing manifest with abstract resources and the plugin manifest, with the actual plugin config.

In particular, we need to make corrections in the following files:
- [ ] https://github.com/docker/docker/blob/master/docs/extend/manifest.md
- [ ] https://github.com/docker/docker/blob/master/docs/extend/index.md
- [ ] https://github.com/docker/docker/blob/master/api/types/plugin.go#L33
- [ ] https://github.com/docker/docker/blob/master/api/types/plugin.go#L143
- [ ] Don't introduce new way of versioning things (`manifestVersion` needs to go)

This must be done before the general release of plugin support.

### PR DESCRIPTION
**Pull request by vieux**
closes https://github.com/docker/docker/issues/27517

@stevvooe @anusha-ragunathan @tiborvass how about this:

`Config -> Settings` it's only set by the user on it's host
`Manifest -> Config` similar to ImageConfig

### CODE REVIEW

**Comment by AkihiroSuda** [comment]
It doesn't compile
```
00:00:54.598 ---> Making bundle: binary-client (in bundles/1.13.0-dev/binary-client)
00:00:55.080 Building: bundles/1.13.0-dev/binary-client/docker-1.13.0-dev
00:01:17.175 # github.com/docker/docker/cli/command/plugin
00:01:17.175 cli/command/plugin/list.go:50: p.Manifest undefined (type *types.Plugin has no field or method Manifest)
```

**Comment by vieux** [comment]
@AkihiroSuda it's not meant to compile, that's why I added [WIP]

it's just a discussion.

**Comment by AkihiroSuda** [comment]
OK :smiley:

**Comment by anusha-ragunathan** [comment]
I like renaming `Manifest` to `Config`.
Regarding rename settings configurable by the user, why not call it `Settings`? Not everything in that list is a host configuration and `HostConfig` is misleading.

**Comment by vieux** [comment]
@anusha-ragunathan I like Settings

**Comment by stevvooe** [comment]
LGTM on s/Config/Settings/ and s/Manifest/Config/. This seems to match what we do for images and clearly identifies the individual components.

**Comment by stevvooe** [comment]
Note that to close #27517, we need to deal with the new way of versioning.

The right approach here is to resolve version through qualified descriptor, rather than embedding.

**Comment by vieux** [comment]
I updated the code, it will compile but still failed since the plugin used in tests aren't updated.

**Comment by thaJeztah** [comment]
In "writing", should we use "configuration" instead of "config"? (Code can still use "config")

**Comment by thaJeztah** [comment]
Design LGTM too, so moving to code review

**Review by thaJeztah — APPROVED** [review]
LGTM

we can make docs changes after code freeze if needed

File: api/types/plugin.go [hunk 1]
```diff
@@ -18,47 +18,26 @@ type Plugin struct {
...
-	// Required: true
-	Manifest PluginManifest `json:"Manifest"`
-
 	// name
 	// Required: true
 	Name string `json:"Name"`
 
+	// settings
+	// Required: true
+	Settings PluginSettings `json:"Settings"`
```
Comment (tiborvass) [inline api/types/plugin.go]: Shouldn't this be pluginconfig since it's manifest being replaced?

**Comment by tiborvass** [comment]
LGTM

**Pull request merged**